core/src/main/java/com/opensymphony/xwork2/ognl/SecurityMemberAccess.java

```diff
@@ -74,11 +81,19 @@ public boolean isAccessible(Map context, Object target, Member member, String pr
             return true;
         }
 
-        if (Modifier.isStatic(member.getModifiers()) && allowStaticMethodAccess) {
-            LOG.debug("Support for accessing static methods [target: {}, targetClass: {}, member: {}, property: {}] is deprecated!",
-                    target, targetClass, member, propertyName);
-            if (!isClassExcluded(memberClass)) {
-                targetClass = memberClass;
+        if (!checkStaticMethodAccess(member, memberModifiers)) {
+            LOG.warn("Access to static method [{}] is blocked!", member);
+            return false;
+        }
+
+        if (isClassExcluded(memberClass)) {
+            LOG.warn("Declaring class of member type [{}] is excluded!", member);
+            return false;
+        } else {
```

Comment:
The else block was introduced (along with moving the `memberClass` check earlier in the call sequence) to try and eliminate two calls to `isClassExcluded(memberClass)` that could happen if the member was static in the original 2.5.x flow.
The only reason to consider/risk the order change was to avoid the 2nd call and use the result of the 1st call to guarantee the implicit state in the else block equivalent to `!isClassExcluded(memberClass)`. 
Walking though the only side-effect appears to be it could potentially fail-out earlier due to the `memberClass` check for cases where it might have also have failed-out due to `isPackageExcluded()` or `isClassExcluded(targetClass)` before ... but for all paths the same false result would be returned (just a different log warning might be seen).  
For that reason it seemed safe-enough to consider/risk to avoid the redundant call.

Reply:
Yes I know why it has been emerged - I meant we were able to drop this piece of code in 2.6 keeping same functionality by reordering checks :)